### ISSUE TITLE
Multiple issues for overflow menu with example code

### ISSUE THREAD

**Issue by nsalyzyn-atimi** [blackberry/jQueryMobile-BB10-Theme#92]
As mentioned in #82, I wanted to share some simple code that would demonstrate several problems we're having using the overflow menu in our project. See below the list I'm providing for the sample code
- _Slow navigation_ - overflow menus that take up more than the full screen scroll up and down surprisingly poor responsiveness.  Touch scrolling up and down on a Dev Alpha B device has a framerate of around 4
- _Default Icon Selected_ - This is more of a question of "how" because it is probably implemented.  In the below example, if you open up the app and select "Page 1", you'll see that the overflow icon now includes "Page 1" and also grabs the icon from the selected menu item (if it had one - which I skipped in this example for simplicity).  How do you make it load the page already open to "Page 1"?
- _Touchstart and Touchend in menu items_ - If you start a touch event inside an overflow menu item and then lift your finger outside of the overflow container, the menu keeps its "selected" status.  Moreover, if you start swiping left and right, you'll find that you can now start scrolling the screen left and right, which can get you into some odd invalid states quite quickly.
- _Scrolling and selecting menu items_ - In the example, given, in addition to the slow navigation when scrolling up and down the overflow menu, you'll also find that it often ends up selecting an item when you're simply trying to scroll up and down.  Once you have a touchstart inside the overflow menu followed by a scroll event, it should disable any possibility for selections. This might partially be due to how I bound events to the menu items - but even if I don't bind any events, the behaviour for when it decides to close the menu is just as awkward.
- _Difficult to bind click events_ - What is the best way of assigning click events to the overflow menu?  "live" vclick events don't seem to bubble up.  You can't add events on pageinit, and on pagechange, you can only access the overflowMenu by doing something like $.mobile.activePage.prevSibling('.overflowMenu').find(...) - that seems unlikely to survive a refactor :)
- _Animations are a mess_ - You'll find in the example if you goto page 2 and back to page 1, the slide animations are a mess.  Slideup is equally as appalling :(.  The overflow menu is visible through much of the transition
- _See through content to overflow_ - in addition to animations, after navigating to page 2 and back to page 1, you may end up seeing through the content through to the overflow menu.  I saw this in my app as well, although I wasn't trying to demonstrate it here, it showed up.

I'm hoping that all of these are fixed in the refactor you're working on.  I started looking into workarounds, but it got messy very quickly.  All these problems can be demonstrated in 100 lines of code and the app I'm writing is a bit larger than that ;)

Thanks in advance for any help you provide!

``` html
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <meta charset="utf-8">
  <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
  <title>Test</title>

  <script>
    var meta = document.createElement("meta");
    meta.setAttribute('name','viewport');
    meta.setAttribute('content','initial-scale='+ (1/window.devicePixelRatio) + ',user-scalable=no');
    document.getElementsByTagName('head')[0].appendChild(meta);
  </script>

  <link rel="stylesheet" href="BlackBerry-JQM-all.min.css" />
  <style>
  </style>

  <script src="BlackBerry-JQM-all.js"></script>
  <script src="local:///chrome/webworks.js" type="text/javascript"></script>

  <script type="text/javascript">
    (function($) {
      $(document).on('pagechange', pageChange);

      function pageChange() {
        // Do it this way beause I cannot use $.mobile.activePage to track down
        // the tabs, unless I use prevSibling()
        $('[data-role="tab"]').each(function() {
          var tab = $(this);
          tab.off('.change-page');
          tab.on('vclick.change-page', function() {
            switch ($(this).text()) {
              case 'Page 1':
                $.mobile.changePage($('#page-1'), { transition : 'slide', reverse : true })
                break;
              case 'Page 2':
                $.mobile.changePage($('#page-2'), { transition : 'slide', reverse : false })
                break;
            }
          })
        })
      }
    })(jQuery)
  </script>
</head>

<body>
  <div data-role="page" id="page-1" data-theme="a">
    <div data-role="header" data-theme="a" data-position="fixed">
      <h1>Page 1</h1>
    </div>

    <div data-role="content">
      <p>Page 1 Content</p>
    </div>

    <div data-role="footer" data-position="fixed">
      <div data-role="actionbar">
        <div data-role="tab" data-overflow="true">Page 1</div>
        <div data-role="tab" data-overflow="true">Page 2</div>
        <div data-role="tab" data-overflow="true">Page 1</div>
        <div data-role="tab" data-overflow="true">Page 2</div>
        <div data-role="tab" data-overflow="true">Page 1</div>
        <div data-role="tab" data-overflow="true">Page 2</div>
        <div data-role="tab" data-overflow="true">Page 1</div>
        <div data-role="tab" data-overflow="true">Page 2</div>
        <div data-role="tab" data-overflow="true">Page 1</div>
        <div data-role="tab" data-overflow="true">Page 2</div>
        <div data-role="tab" data-overflow="true">Page 1</div>
        <div data-role="tab" data-overflow="true">Page 2</div>
        <div data-role="tab" data-overflow="true">Page 1</div>
        <div data-role="tab" data-overflow="true">Page 2</div>
        <div data-role="tab" data-overflow="true">Page 1</div>
      </div>
    </div>

  </div>

  <div data-role="page" id="page-2" data-theme="a">
    <div data-role="header" data-theme="a" data-position="fixed">
      <h1>Page 2</h1>
    </div>

    <div data-role="content">
      <p>Page 2 Content</p>
    </div>

    <div data-role="footer" data-position="fixed">
      <div data-role="actionbar">
        <div data-role="tab" data-overflow="true">Page 1</div>
        <div data-role="tab" data-overflow="true">Page 2</div>
        <div data-role="tab" data-overflow="true">Page 1</div>
        <div data-role="tab" data-overflow="true">Page 2</div>
        <div data-role="tab" data-overflow="true">Page 1</div>
        <div data-role="tab" data-overflow="true">Page 2</div>
        <div data-role="tab" data-overflow="true">Page 1</div>
        <div data-role="tab" data-overflow="true">Page 2</div>
        <div data-role="tab" data-overflow="true">Page 1</div>
        <div data-role="tab" data-overflow="true">Page 2</div>
        <div data-role="tab" data-overflow="true">Page 1</div>
        <div data-role="tab" data-overflow="true">Page 2</div>
        <div data-role="tab" data-overflow="true">Page 1</div>
        <div data-role="tab" data-overflow="true">Page 2</div>
        <div data-role="tab" data-overflow="true">Page 1</div>
      </div>
    </div>

  </div>
</body>
</html>
```

**Comment by nsalyzyn-atimi** [comment]
Somebody in the office here said he got it running more smoothly by adding -webkit-overflow-scrolling: touch; to the overflow menu.  However, we've been getting this "white bar" artifact that occurs occasionally that becomes consistent after adding that css. In either case, it might be a suggested fix if you're having trouble with poor scrolling still.

Hope that helps,
Nick

**Comment by nsalyzyn-atimi** [comment]
Additional bug to be tested:
- When overflow menu is open, you can still interact with the main page.  This can make it difficult to close the overflow menu.

**Comment by Gerii** [comment]
There's already a pull request for this: https://github.com/blackberry/jQueryMobile-BB10-Theme/pull/79

**Comment by nsalyzyn-atimi** [comment]
That pull request captures the case where you want to close the menu by clicking on the overflow menu button.  Which is definitely an improvement  However, it doesn't capture the case where you have links on the side of the screen - that whole right-side area should close the overflow menu rather than being clickable, correct?  Some sort of overlay.

**Comment by Gerii** [comment]
If it really fixes the issue referenced it should close it when tapping anywhere on the screen except for the buttons of the overflow menu.

**Comment by nsalyzyn-atimi** [comment]
Posted a comment on the pull request, since code review clearly illustrates what is still outstanding. In any case, the overflow menu is going through a refactor, so most likely the pull request will be invalid in a week or two.

**Comment by jasondscott** [comment]
@nsalyzyn-atimi @Gerii #96 should fix the ability to interact with the page. Let me know if you still see this issue.

**Comment by nsalyzyn-atimi** [comment]
Thanks for the partial fix :)  Looking forward to seeing your refactor.

**Comment by nsalyzyn-atimi** [comment]
@jasondscott #96 did not fix the issue. :(  It was a partial partial fix, since tapping on the overflow menu icon often closes the overflow menu - but tapping anywhere else usually taps through to the item below.

**Comment by jasondscott** [comment]
@nsalyzyn-atimi @Gerii The action bar refactor is up in a branch https://github.com/blackberry/jQueryMobile-BB10-Theme/tree/actionbar-refactor 
The action bar is pretty much done.  The overflow needs some work still before we can land this in master.
Check it out, feedback is appreciated. 

Thanks

**Comment by jasondscott** [comment]
Here is a sample page https://github.com/blackberry/jQueryMobile-BB10-Theme/blob/actionbar-refactor/kitchenSink/actionbar_sample.html

**Comment by lumierephoto** [comment]
@jasondscott  The action bar refactor sample uses external html pages when changing pages. Is that necessary, or could I use # tags for a one page app?

**Comment by jasondscott** [comment]
There should be no problem with using internal pages. This sample is just using $.mobile.changepage which works with internal pages. Sent from my BlackBerry 10 smartphone.From: lumierephotoSent: Saturday, January 12, 2013 3:48 PMTo: blackberry/jQueryMobile-BB10-ThemeReply To: blackberry/jQueryMobile-BB10-ThemeCc: Jason ScottSubject: Re: [jQueryMobile-BB10-Theme] Multiple issues for overflow menu with
 example code (#92)@jasondscott  The action bar refactor sample uses external html pages when changing pages. Is that necessary, or could I use # tags for a one page app?

```
          —
          Reply to this email directly or view it on GitHub.
```

**Comment by lumierephoto** [comment]
@jasondscott Cool thanks! Just checking.

Also, I'm assuming that the refactored code is in the BlackBerry-JQM-all.js and the BlackBerry-JQM-all.css files? 

I'd like to test it out.

**Comment by lumierephoto** [comment]
@jasondscott  From some quick testing it's working in the browser, but in the simulator it's going even more wacky than before. And by wacky, it will change to the wrong page, then when I hit the back button on that wrong page, the right page it should have changed to will be there. Again, this is not happening in the browser. 

Also, the footer with the actionbar tabs are not rendering correctly in either the browser or simulator. Even when I added data-position="fixed". They are fixing to the div above, which in my case is the data-role="content" div.

When I put links to different pages in the Overflow menus, both the right and left panels like this for example:

```
        <div data-role="panel" id="left" data-position="left" data-display="reveal" data-theme="a">
            <ul data-role="listview" data-theme="a">
                <li><a href="#Acura">Acura</a></li>
                <li><a href="#Audi">Audi</a></li>
                <li><a href="#Bmw">BMW</a></li>
            </ul>
        </div>
        <div data-role="panel" id="right" data-position="right" data-display="overlay" data-theme="a">
            <div>
                <ul data-role="listview" data-theme="a">
                    <li><a href="#Acura">Acura</a></li>
                    <li><a href="#Audi">Audi</a></li>
                    <li><a href="#Bmw">BMW</a></li>
                </ul>
            </div>
        </div>
```

It will go to the right page. However, when I navigate back to the home page where I have the actionbar implemented, and I were to open up either the right or left panel with the tabs in the action bar, it won't open. What will happen is that there will be "cover" transition on the home page transitioning to itself. 

In this test, I'm pointing to the refactored BlackBerry-JQM-all.js and the BlackBerry-JQM-all.css files. As well, I've implement the code like here:  https://github.com/blackberry/jQueryMobile-BB10-Theme/blob/actionbar-refactor/kitchenSink/actionbar_sample.html

**Comment by jasondscott** [comment]
All of these issues should be fixed by #100 
If not we can reopen.
Closing.
